include/ginkgo/core/factorization/par_ilu.hpp

```diff
@@ -157,7 +171,17 @@ class ParIlu : public Composition<ValueType> {
         : Composition<ValueType>(factory->get_executor()),
           parameters_{factory->get_parameters()}
     {
-        generate_l_u(system_matrix, parameters_.skip_sorting)->move_to(this);
+        if (parameters_.l_strategy == nullptr) {
+            parameters_.l_strategy =
+                std::make_shared<typename l_matrix_type::classical>();
```

Comment:
Wouldn't `automatical` provide better generality?

Reply:
Exactly, I was thinking about using `automatical` here, but it would not work if the Executor is Reference or Omp. I was thinking about using `dynamic_cast` to figure it out, however, I think that would be a lot better if that was done inside `automatical` itself.
So for now, I simply use `classical` because it does not require information about the executor and can also handle multiple right hand sides (in contrast to the previously used `sparselib`).